src/pool/spawn.rs

```diff
@@ -120,6 +125,61 @@ impl<T> QueueCore<T> {
             }
         }
     }
+
+    /// Scale workers.
+    pub fn scale_workers(&self, new_thread_count: usize) -> bool {
```

Comment:
Maybe we'd better protect the whole function with a `Mutex`? I'm afraid the status would be strange if this function is called concurrently in multiple threads.

Reply:
It has been added, but I think it is better to be controlled by a management thread in actual use. Do we need annotations?

Reply:
It is better to guarantee thread safety of public functions if possible.

Can you please explain the reason why it should be controlled by a management thread? I think we can assume this function won't be called frequently, so such a `Mutex` will not cause performance issues in most cases.

Reply:
Oh, I thought that using multiple threads to scale threads might feel a bit weird. But I agree with you,  as much as possible to ensure that the pub interface is thread-safe. I'll modify `Mutex`'s place as your another advice. Thanks!